### ISSUE TITLE
Version 0.8.10 appears to be broken

### ISSUE THREAD

**Issue by rowillia** [iziz/libPhoneNumber-iOS#108]
Looking at the CI results, it appears that 0.8.10 doesn't compile.

https://github.com/iziz/libPhoneNumber-iOS/compare/0.8.8...0.8.10

When CocoaPods picked up the latest version this broke our build.

**Comment by iziz** [comment]
Please show me your error messages. (I'm waiting for supporting swift 2.1 on travis CI)
And "And set the Embedded Content Contains Swift to "Yes" in your build settings." as documented.

**Comment by jclardy** [comment]
I am getting a build error too, I am guessing related to the swift changes. 

"Use of '@import' when modules are disabled"

**Comment by jclardy** [comment]
Fixed my issue with "Enabled Modules (C and Objective-C)" which was set to no on our project. Can the cocoapods config set that automatically somehow?

**Comment by frodella-daitan** [comment]
I'm having the same `@import` error and `Enable Modules` is set to YES in my project and all the pod targets as well. Any clues?

**Comment by mihaibabusca** [comment]
Are including the .h file in a .mm file ?

**Comment by frodella-daitan** [comment]
Hmm, I do have a single .mm file in my project, but I just imported the libPhoneNumber headers from the bridging header, since the class that will use it is written in Swift and I can't use frameworks since our app supports iOS 7.

**Comment by iziz** [comment]
Did you set the Embedded Content Contains Swift to "Yes" in your build settings ?

**Comment by frodella-daitan** [comment]
Yes, it wasn't but then I switched it to `Yes` and then error persisted. Also, I don't see any Swift sources in the Pod source folder.

**Comment by iziz** [comment]
[TestJustLib.zip](https://github.com/iziz/libPhoneNumber-iOS/files/62430/TestJustLib.zip)
Here is sample project. But I can't find any problems

**Comment by frodella-daitan** [comment]
Indeed... I took your project and added a Swift class, importing the libPhoneNumber's headers from the bridging header, and it worked fine. I'll try to compare it with my project to see if I can narrow down the cause, and I'll write back if I find anything.

Thanks for looking into it!

**Comment by iziz** [comment]
OK, thanks!

**Comment by frodella-daitan** [comment]
After some debugging, it seems my case comes down to the following combination of conditions:
- Mixed Swift/Obj-C project
- libPhoneNumber headers added to the bridging header
- .mm (Obj-C++) source present in the project, which imports `<project>-Swift.h`

I took a look at my generated `-Swift.h` header file and it imports the bridging header. Since Obj-C++ does not support modules and my .mm file imports the `-Swift.h` header, that error is generated (@mihaibabusca was right on the money).

Since this is a very particular situation, I'll see if I can work around it somehow in my project.

**Comment by gautier-gdx** [comment]
@frodella-daitan  Did you find a solution? I have the exact same condition!
 I have also tried to switch to carthage it did work but i couldn't ship it on iTunes connect having another error related to libPhoneNumber too. so carthage is not an option for me right now.

anyway I will post an issue about this carthage's error when I will be sure I'm not the problem

**Comment by frodella-daitan** [comment]
@gautier-gdx I was able able to refactor my code so that the libPhoneNumber headers were not included in Obj-C++ files. Other than that, I'm not really sure what else could be done...
